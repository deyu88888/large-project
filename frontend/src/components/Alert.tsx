import MuiAlert, { AlertProps } from "@mui/material/Alert";

interface CustomAlertProps extends AlertProps {}

export const Alert = ({ children, ...props }: CustomAlertProps) => {
  return (
    <MuiAlert elevation={6} variant="filled" {...props}>
      {children}
    </MuiAlert>
  );
};
