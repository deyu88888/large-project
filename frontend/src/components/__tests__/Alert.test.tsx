import React from 'react';
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Alert } from '../Alert';

describe('Alert Component', () => {
  it('renders the alert with default props', () => {
    const { getByRole } = render(<Alert>Test Alert</Alert>);
    
    const alertElement = getByRole('alert');
    expect(alertElement).toBeTruthy();
    expect(alertElement.textContent).toBe('Test Alert');
  });

  it('passes through additional MUI Alert props', () => {
    const { getByRole } = render(
      <Alert severity="error" color="error">
        Error Alert
      </Alert>
    );
    
    const alertElement = getByRole('alert');
    expect(alertElement).toHaveClass('MuiAlert-filledError');
  });

  it('matches snapshot with default props', () => {
    const { container } = render(<Alert>Snapshot Alert</Alert>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const { getByText } = render(
      <Alert>
        <span>Custom Content</span>
      </Alert>
    );
    
    const childElement = getByText('Custom Content');
    expect(childElement).toBeTruthy();
  });

  it('has correct default MuiAlert props', () => {
    const { container } = render(<Alert>Default Props Alert</Alert>);
    const alertElement = container.firstChild as HTMLElement;
    
    expect(alertElement).toHaveClass('MuiAlert-filled');
    expect(alertElement).toHaveClass('MuiPaper-elevation6');
  });
});