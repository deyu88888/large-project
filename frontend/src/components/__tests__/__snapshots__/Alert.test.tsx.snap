// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > matches snapshot with default props 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 MuiAlert-root MuiAlert-colorSuccess MuiAlert-filledSuccess MuiAlert-filled css-1t1l72a-MuiPaper-root-MuiAlert-root"
  role="alert"
  style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);"
>
  <div
    class="MuiAlert-icon css-vab54s-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1ckov0h-MuiSvgIcon-root"
      data-testid="SuccessOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-zioonp-MuiAlert-message"
  >
    Snapshot Alert
  </div>
</div>
`;
